pkg/gatherers/clusterconfig/gather_nodes_test.go

```diff
@@ -0,0 +1,199 @@
+			assert.Equal(t, tt.wantRecords, records)
+			assert.Len(t, errs, tt.wantErrsCount)
+		})
+	}
+}
+
+func Test_isProductNamespacedKey(t *testing.T) {
+	tests := []struct {
+		name string
+		key  string
+		want bool
+	}{
+		{
+			name: "contains openshift.io/",
+			key:  "openshift.io/something",
+			want: true,
+		},
+		{
+			name: "contains k8s.io/",
+			key:  "k8s.io/something",
+			want: true,
+		},
+		{
+			name: "contains kubernetes.io/",
+			key:  "kubernetes.io/something",
+			want: true,
+		},
+		{
+			name: "does not contain anything",
+			key:  "something",
+			want: false,
+		},
+	}
+
+	for _, tt := range tests {
+		result := isProductNamespacedKey(tt.key)
+		assert.Equal(t, tt.want, result)
+	}
+}
+
+func Test_isRegionLabel(t *testing.T) {
+	tests := []struct {
+		name string
+		key  string
+		want bool
+	}{
+		{
+			name: "key with 'failure-domain.beta.kubernetes.io/region'",
+			key:  "failure-domain.beta.kubernetes.io/region",
+			want: true,
+		},
+		{
+			name: "key with 'topology.kubernetes.io/region'",
+			key:  "topology.kubernetes.io/region",
+			want: true,
+		},
+		{
+			name: "key without 'failure-domain.beta.kubernetes.io/region' or 'topology.kubernetes.io/region'",
+			key:  "other/key",
+			want: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := isRegionLabel(tt.key)
+			assert.Equal(t, tt.want, result)
+		})
+	}
+}
+
+func Test_anonymizeNode(t *testing.T) {
+	tests := []struct {
+		name string
+		node *corev1.Node
+		want *corev1.Node
+	}{
+		{
+			name: "successful anonymize node",
+			node: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "test-node",
+					Annotations: map[string]string{"annot-key": "annot-value"},
+					Labels:      map[string]string{"label-key": "label-value"},
+				},
+				Status: corev1.NodeStatus{
+					NodeInfo: corev1.NodeSystemInfo{
+						BootID:          "boot-id",
+						SystemUUID:      "system-uuid",
+						MachineID:       "machine-id",
+						OperatingSystem: "operating-system",
+						Architecture:    "architecture",
+					},
+					Images: []corev1.ContainerImage{
+						{
+							Names:     []string{"image-name"},
+							SizeBytes: 123,
+						},
+					},
+				},
+			},
+			want: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "test-node",
+					Annotations: map[string]string{"annot-key": ""},
+					Labels:      map[string]string{"label-key": anonymize.String("label-value")},
+				},
+				Status: corev1.NodeStatus{
+					NodeInfo: corev1.NodeSystemInfo{
+						BootID:          anonymize.String("boot-id"),
+						SystemUUID:      anonymize.String("system-uuid"),
+						MachineID:       anonymize.String("machine-id"),
+						OperatingSystem: "operating-system",
+						Architecture:    "architecture",
+					},
+					Images: nil,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equal(t, tt.want, anonymizeNode(tt.want), tt.want)
```

Comment:
I think this should be `assert.Equal(t, tt.want, anonymizeNode(tt.node))`, right?

Reply:
I guess that I wanted too much here. LOL